programs/vote/src/vote_state/mod.rs

```diff
@@ -36,9 +38,69 @@ pub const INITIAL_LOCKOUT: usize = 2;
 // Maximum number of credits history to keep around
 pub const MAX_EPOCH_CREDITS_HISTORY: usize = 64;
 
-// Offset of VoteState::prior_voters, for determining initialization status without deserialization
+// Offset of VoteState::pri : Clone + Debug {or_voters, for determining initialization status without deserialization
 const DEFAULT_PRIOR_VOTERS_OFFSET: usize = 82;
 
+// VoteTransactionClone hack is done so that we can derive clone on the tower that uses the
+// VoteTransaction trait object. Clone doesn't work here because it returns Self which is not
+// allowed for trait objects
+#[typetag::serde{tag = "type"}]
+pub trait VoteTransaction: VoteTransactionClone + Debug + Send {
+    fn slot(&self, i: usize) -> Slot;
+    fn len(&self) -> usize;
+    fn hash(&self) -> Hash;
+    fn timestamp(&self) -> Option<UnixTimestamp>;
+    fn last_voted_slot(&self) -> Option<Slot>;
+    fn last_voted_slot_hash(&self) -> Option<(Slot, Hash)>;
+    fn set_timestamp(&mut self, ts: Option<UnixTimestamp>);
+
+    fn slots(&self) -> Vec<Slot> {
+        (0..self.len()).map(|i| self.slot(i)).collect()
+    }
+
+    fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    // Have to manually implement because deriving PartialEq returns Self
+    fn eq(&self, other: &dyn VoteTransaction) -> bool;
+    fn as_any(&self) -> &dyn Any;
+}
+
+pub trait VoteTransactionClone {
+    fn clone_box(&self) -> Box<dyn VoteTransaction>;
+}
+
+impl<T> VoteTransactionClone for T
+where
+    T: VoteTransaction + Clone + 'static,
+{
+    fn clone_box(&self) -> Box<dyn VoteTransaction> {
+        Box::new(self.clone())
+    }
+}
+
+impl Clone for Box<dyn VoteTransaction> {
+    fn clone(&self) -> Box<dyn VoteTransaction> {
+        self.clone_box()
+    }
+}
```

Comment:
Heh, this is the standard way to work around the Trait object/Clone incompatibility